devito/operator/operator.py

```diff
@@ -956,6 +956,7 @@ def parse_kwargs(**kwargs):
     options.setdefault('cire-mincost-sops', opt_options.get('cire-mincost-sops', None))
     options.setdefault('openmp', openmp)
     options.setdefault('mpi', configuration['mpi'])
+    options.setdefault('gpu-direct', False)
```

Comment:
after #1387 went in, this should now be moved somewhere around [here](https://github.com/devitocodes/devito/blob/master/devito/core/gpu_openmp.py#L266)

Reply:
It seems like it is also necessary to add the gpu-direct somewhere else. By just moving it from `operator.py` to `gpu_openmp.py` leads to the following exception:
`devito.exceptions.InvalidOperator: Unsupported optimization options: [gpu-direct]`

Reply:
a `git grep "Unsupported optimization options" shows that that exception is only raised in one point, here:

https://github.com/devitocodes/devito/blob/master/devito/core/gpu_openmp.py#L268

it happens because you have to `pop` `'gpu-direct'` from `oo`, so that devito can raise an exception if an illegal keyword has been provided. So `pop` is the way to go, just like done with all other options

Reply:
this line should be removed as per my comment above